src/renderer/dx/DxRenderer.hpp

```diff
@@ -221,12 +242,20 @@ namespace Microsoft::Console::Render
         // DirectX constant buffers need to be a multiple of 16; align to pad the size.
         __declspec(align(16)) struct
         {
-            float ScaledScanLinePeriod;
-            float ScaledGaussianSigma;
+            // Note: This can be seen as API endpoint towards user provided pixel shaders.
+            //  Changes here can break existing pixel shaders so be careful with changing datatypes
+            //  and order of parameters
+            float Time;
+            float Scale;
+            DirectX::XMFLOAT2 Resolution;
+            DirectX::XMFLOAT4 Background;

```

Comment:
TODO: Can this struct be safely extended with more attributes later on as long as we add them at the end?

Reply:
For the future: Shaders could benefit from custom parameters as well. No idea how to do it just now.